lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -820,22 +819,22 @@ joinStakePool ctx spl (ApiT poolId) (ApiT wid) passwd = do
     (_, walMeta, _) <- liftHandler $ withWorkerCtx ctx wid throwE $
         \wrk -> W.readWallet wrk wid
     when (walMeta ^. #delegation == W.Delegating poolId) $
-        liftHandler $ throwE (ErrCreateCertPoolAlreadyJoined poolId)
+        liftHandler $ throwE (ErrSelectForDelegationPoolAlreadyJoined poolId)
 
     allPools <- listPools spl
     when (poolIsUnknown allPools) $
-        liftHandler $ throwE (ErrCreateCertNoSuchPool poolId)
+        liftHandler $ throwE (ErrSelectForDelegationNoSuchPool poolId)
 
     let (ApiWalletPassphrase (ApiT pwd)) = passwd
 
     selection <- liftHandler $ withWorkerCtx ctx wid liftE1 $ \wrk ->
-        W.createCert @_ @s @t wrk wid
+        W.selectCoinsForDelegation @_ @s @t wrk wid
 
     (tx, meta, time, wit) <- liftHandler $ withWorkerCtx ctx wid liftE2 $ \wrk ->
-        W.signCert @_ @s @t @k wrk wid () pwd selection poolId
+        W.signDelegation @_ @s @t @k wrk wid () pwd selection poolId
```

Comment:
I'm wondering if we don't have a potential race-condition with `POST /transaction` and `joinStakePool`:

We run:
1. `selectCoinsForDelegation` does the coin selection based on the `availableUtxo`.
2. `signDelegation`
3. `submitTx` — modifies the `availableUtxo` by adding a pending transaction

Step 1 and 3 are run in different DB transactions.

So if we select coins for another tx, while the first tx has not yet been submitted, we could spend the same funds twice.

Reply:
Indeed, but that's nothing new :) 
We already have that particular race condition from making two concurrent payments in parallel. Worst that can happen is that we create and sign both transactions, we submit both, and only one get through in the end while the other stay pending forever.

Reply:
> we could end up spending the same funds twice.

No! The network will prevent that!

Reply:
> No! The network will prevent that!

🙄 Ok, bad wording on my part